compiler/src/dotty/tools/dotc/semanticdb/Scala3.scala

```diff
@@ -126,6 +126,85 @@ object Scala3:
         sym.is(Synthetic) && !sym.isAnonymous && !sym.name.isEmptyNumbered
 
+      def symbolInfo(symkinds: Set[SymbolKind])(using LinkMode, Context, SemanticSymbolBuilder, TypeOps): SymbolInformation =
+        val sname = sym.symbolName
+        val signature = sym.info.toSemanticSig(sym)
+        SymbolInformation(
+          symbol = sname,
+          language = Language.SCALA,
+          kind = symbolKind(symkinds),
+          properties = sym.symbolProps(symkinds),
+          displayName = Symbols.displaySymbol(sym),
+          signature = signature,
+        )
+
+      private def symbolKind(symkinds: Set[SymbolKind])(using Context): SymbolInformation.Kind =
+        if sym.isTypeParam then
+          SymbolInformation.Kind.TYPE_PARAMETER
+        else if sym.is(TermParam) then
+          SymbolInformation.Kind.PARAMETER
+        else if sym.isTerm && sym.owner.isTerm then
+          SymbolInformation.Kind.LOCAL
+        else if sym.isInlineMethod || sym.is(Macro) then
+          SymbolInformation.Kind.MACRO
+        else if sym.isConstructor then
+          SymbolInformation.Kind.CONSTRUCTOR
+        else if sym.isSelfSym then
+          SymbolInformation.Kind.SELF_PARAMETER
+        else if sym.isOneOf(Method) || symkinds.exists(_.isVarOrVal) then
+          SymbolInformation.Kind.METHOD
+        else if sym.isPackageObject then
+          SymbolInformation.Kind.PACKAGE_OBJECT
+        else if sym.is(Module) then
+          SymbolInformation.Kind.OBJECT
+        else if sym.is(Package) then
+          SymbolInformation.Kind.PACKAGE
+        else if sym.isAllOf(JavaInterface) then
+          SymbolInformation.Kind.INTERFACE
+        else if sym.is(Trait) then
+          SymbolInformation.Kind.TRAIT
+        else if sym.isClass then
+          SymbolInformation.Kind.CLASS
+        else if sym.isType then
+          SymbolInformation.Kind.TYPE
+        else if sym.is(ParamAccessor) then
+          SymbolInformation.Kind.FIELD
+        else
+          SymbolInformation.Kind.UNKNOWN_KIND
+
+      private def symbolProps(symkinds: Set[SymbolKind])(using Context): Int =
+        if sym.is(ModuleClass) then
+          return sym.sourceModule.symbolProps(symkinds)
+        var props = 0
+        if sym.isPrimaryConstructor then
+          props |= SymbolInformation.Property.PRIMARY.value
+        if sym.is(Abstract) || symkinds.contains(SymbolKind.Abstract) then
+          props |= SymbolInformation.Property.ABSTRACT.value
+        if sym.is(Final) then
+          props |= SymbolInformation.Property.FINAL.value
+        if sym.is(Sealed) then
+          props |= SymbolInformation.Property.SEALED.value
+        if sym.isOneOf(GivenOrImplicit) then
+          props |= SymbolInformation.Property.IMPLICIT.value
+        if sym.is(Lazy, butNot=Module) then
+          props |= SymbolInformation.Property.LAZY.value
+        if sym.isAllOf(Case | Module) || sym.is(CaseClass) || sym.isAllOf(EnumCase) then
```

Comment:
`EnumCase` how we identify singleton enum values, so `case` should be there as it is part of the definition

Reply:
Case is defined as: `CASE | Has a case modifier?`

So in reality, it doesn't have a case modifier, it's more akin to match cases. I think it would be best to have it as a EnumCase, separate from just Case

Reply:
Im not sure what you mean,
```scala
enum Color {
  case Red
  ...
}
```

here `Red` is defined with the `case` modifier

Reply:
Is it a modifier though? With case class it is indeed a modifier, in enum those cases are obligatory, without case they do not exist. You cannot write just "Red"

Reply:
I think it is important, `case` distinguishes `Red` from being any old val (like its desugared form), which plays into things like exhaustivity checking or the "provably disjoint" check for match types. I do see we do give them the `enum` modifier also at least.

Reply:
Ok, I guess if we have both it is an EnumCase kind. Makes sense!

Reply:
I think it's ok to add `Case` to EnumCase, since we can tell the symbols of EnumCase by checking if there're properties : `Enum` and `Case`